### PR TITLE
updated the upload_results doc to more accurately describe its input and output

### PR DESCRIPTION
**Pull request by grant-Kolena**
### Linked issue(s):

Screenshot 2024-01-15 at 1 54 01 PM

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/dataset/evaluation.py [hunk 1]
```diff
@@ -169,8 +169,11 @@ def upload_results(
     :param dataset: The name of the dataset.
     :param model: The name of the model.
     :param results: Either a DataFrame or a list of tuples, where each tuple consists of
-                    an eval configuration and a DataFrame.
-    :return: None
+                    an eval configuration and a DataFrame. The DataFrame type has been aliased to be either a pandas
+                    DataFrame or an iterator of pandas DataFrame, this allows for batch upload when passing iterators.
```
Comment (munkyshi) [inline kolena/dataset/evaluation.py]: I'd prefer to link to the alias for `DataFrame` that we have + have the description there.

File: kolena/dataset/evaluation.py [hunk 2]
```diff
@@ -169,8 +169,11 @@ def upload_results(
...
+
+    :return: An UploadResultsResponse object containing information about the number of result rows created
+             and updated
```
Comment (munkyshi) [inline kolena/dataset/evaluation.py]: It's not very meaningful to refer to `UploadResultsResponse` when we don't have the type documented -- how will users know what the fields to look into actually are?

File: kolena/dataset/evaluation.py [hunk 3]
```diff
@@ -169,8 +169,11 @@ def upload_results(
     :param dataset: The name of the dataset.
     :param model: The name of the model.
     :param results: Either a DataFrame or a list of tuples, where each tuple consists of
-                    an eval configuration and a DataFrame.
-    :return: None
+                    an eval configuration and a DataFrame. The DataFrame type has been aliased to be either a pandas
```
Comment (ly-kolena) [inline kolena/dataset/evaluation.py]: Is there a way for `DataFrame` type to be expanded or linkable to the source definition? Exaplaining type alias seems a bit verbose.

Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: currently there is a not very descriptive tooltip, seems only the python builtin types gets a link
Screenshot 2024-01-12 at 11 28 22 AM

File: kolena/dataset/evaluation.py (hunk 2 above)
Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: I will add the doc for it

**Comment by grant-Kolena** [comment]
We have description linked for every self defined type now
Screenshot 2024-01-16 at 10 14 25 AM

File: kolena/dataset/evaluation.py (hunk 3 above)
Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: They are expandable now

File: kolena/dataset/evaluation.py (hunk 2 above)
Comment (gordonhart) [reply kolena/dataset/evaluation.py]: This actually raises a larger problem: `UploadResultsResponse` is an internal API object and shouldn't leak into the external-facing API as the response from `upload_results`.

What are users expected to do with an `UploadResultsResponse`? We should consider changing this function to return `None`.

Comment (ly-kolena) [reply kolena/dataset/evaluation.py]: @gordonhart What would you recommend if we want to make `UploadResultsResponse` public?

Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: Or can we create a public data model, I still think number of rows created/ updated is valuable info

File: kolena/dataset/evaluation.py [hunk 4]
```diff
@@ -197,12 +196,33 @@ def upload_results(
...
                     id_column_validated = True
+                total_rows += df_result.shape[0]
                 df_results = _process_result(config, df_result, id_fields)
                 upload_data_frame(df=df_results, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
 
-    response = _upload_results(model, load_uuid, existing_dataset.id)
+    response = _send_upload_results_request(model, load_uuid, existing_dataset.id)
     log.info(
         f"uploaded test results for model '{model}' on dataset '{dataset}': "
+        f"There are a total of {total_rows} rows in the results input"
```
Comment (ly-kolena) [inline kolena/dataset/evaluation.py]: Prefer to keep log succinct, suggest `{} uploaded, {} inserted, {} updated`

Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: sounds good

Comment (ly-kolena) [reply kolena/dataset/evaluation.py]: Do we also want to include result link to web UI?

File: kolena/dataset/evaluation.py [hunk 5]
```diff
@@ -51,7 +51,14 @@
 from kolena.errors import NotFoundError
 
 EvalConfig = Optional[Dict[str, Any]]
+"""
+User defined configuration for evaluating results, for example `{"threshold": 7}`.
+"""
 DataFrame = Union[pd.DataFrame, Iterator[pd.DataFrame]]
+"""
+A type alias representing a DataFrame, which can be either a pandas DataFrame
+or an iterator of pandas DataFrames, data will be processed in chunks when iterator is used.
```
Comment (ly-kolena) [inline kolena/dataset/evaluation.py]: Is `data will be processed...` needed? Seems `iterator` by itself is sufficient. (side nit, `in chunks` of iterator is confusing)

File: kolena/dataset/evaluation.py (hunk 4 above)
Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: We do have a task of logging the result url, do you mean we should return the result url in the response?

File: kolena/dataset/evaluation.py (hunk 5 above)
Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: how about we change it to `A type alias representing a DataFrame, which can be either a pandas DataFrame
or an iterator of pandas DataFrames`?

File: kolena/dataset/evaluation.py (hunk 4 above)
Comment (ly-kolena) [reply kolena/dataset/evaluation.py]: I meant only logging the result url, fine for the other ticket to cover it.

File: kolena/dataset/evaluation.py (hunk 5 above)
Comment (ly-kolena) [reply kolena/dataset/evaluation.py]: Sounds good.

File: kolena/dataset/evaluation.py (hunk 4 above)
Comment (grant-Kolena) [reply kolena/dataset/evaluation.py]: I still think this is valuable to see `{} uploaded, {} inserted, {} updated`. I don't know a good way to get these info for a result file upload from sdk on the UI, plus this is one less click

**Review by ly-kolena — APPROVED** [review]

**Pull request merged**